packages/jellyfish-api-core/__tests__/category/icxorderbook/createOrder.test.ts

```diff
@@ -0,0 +1,174 @@
+        vout: utxo.vout
+      }
+    })
+    const result: ICXGenericResult = await client.icxorderbook.createOrder(order, inputUTXOs)
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders
+    const orders: Record<string, ICXOrderInfo| ICXOfferInfo> = await container.call('icx_listorders', [])
+    // we know that only ICXOrderInfo will be returned, so cast and pass to check order details
+    await checkDFISellOrderDetails(container, order, createOrderTxId, orders as Record<string, ICXOrderInfo>)
+
+    // check accountDFI[idDFI] balance, reduced by 15 DFI
+    const accountDFIAfterOrder = await container.call('getaccount', [accountDFI, {}, true])
+    expect(Number(accountDFIAfterOrder[idDFI])).toStrictEqual(Number(accountDFIStart[idDFI] - Number(15)))
+  })
+
+  it('Should return an error when invalid ICXOrder.tokenFrom is used', async () => {
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: 'DOGE',
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const promise = client.icxorderbook.createOrder(order, [])
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Token DOGE does not exist!\', code: -8, method: icx_createorder')
+  })
+
+  it('Should return an error when invalid ICXOrder.chainTo is used', async () => {
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'DOGE',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const promise = client.icxorderbook.createOrder(order, [])
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Invalid parameters, argument "chainTo" must be "BTC" if "tokenFrom" specified\', code: -8, method: icx_createorder')
+  })
+
+  it('Should return an error when invalid ICXOrder.ownerAddress is used', async () => {
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: '123',
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const promise = client.icxorderbook.createOrder(order, [])
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'recipient (123) does not refer to any valid address\', code: -5, method: icx_createorder')
+  })
+
+  it('Should return an error when invalid ICXOrder.receivePubkey is used', async () => {
+    // create order - maker
```

Comment:
```suggestion
  it('should create order - maker and return an error when invalid ICXOrder.receivePubkey is used', async () => {
```

Reply:
"nitty gritty" don't use captilize `should` as semantically within JS ecosystem BDD tests are usually lower case